test/unit/test_sproxyd_client.py

```diff
@@ -242,27 +242,108 @@ def test_del_object(self, mock_http, _):
 
         mock_http.assert_called_once_with('DELETE',
                                           'http://host:81/path/obj_1',
-                                          headers=None, preload_content=False)
+                                          headers=None, body=None, preload_content=False)
 
     def test_get_object(self):
-        server = eventlet.listen(('127.0.0.1', 0))
-        (ip, port) = server.getsockname()
-
         content = 'Hello, World!'
 
-        def hello_world(env, start_response):
+        def my_app(env, start_response):
             start_response('200 OK', [('Content-Type', 'text/plain')])
             return [content]
 
-        t = eventlet.spawn(eventlet.wsgi.server, server, hello_world)
+        with utils.WSGIServer(my_app) as server:
+            with mock.patch('eventlet.spawn'):
+                sproxyd_client = SproxydClient(['http://%s:%d/path'
+                                                % (server.ip, server.port)])
+            obj = sproxyd_client.get_object('ignored')
 
-        sproxyd_client = SproxydClient(['http://%s:%d/path' % (ip, port)])
-        obj = sproxyd_client.get_object('ignored')
+            self.assertEqual(content, obj.next())
+            # Assert that `obj` is an Iterable
+            self.assertRaises(StopIteration, obj.next)
 
-        self.assertEqual(content, obj.next())
-        # Assert that `obj` is an Iterable
-        self.assertRaises(StopIteration, obj.next)
-        t.kill()
+    def test_put_object_with_metadata(self):
+        obj_name = utils.get_random_unicode(10).encode('utf-8')
+        obj_metadata = utils.get_random_unicode(128).encode('utf-8')
+        obj_body = utils.get_random_unicode(1024).encode('utf-8')
+
+        def my_app(env, start_response):
+            assert 'PUT' == env['REQUEST_METHOD']
```

Comment:
It's looks weird to have reference to 'self' in a function (ie not a method) that will be executed inside a greenthread launched by in another file (utils.py). I can see myself asking in a few month from now : "Who the f* is 'self' here" ?